### PR TITLE
Deploy greenhouse and docker-mirror-proxy to new control plane cluster

### PR DESCRIPTION
**Pull request by brianmcarey**
Greenhouse(Bazel cache) and the docker-mirror-proxy are required on the new cluster in order to enable the migration of workloads from the existing control plane cluster.

/cc @xpivarc @dhiller @enp0s3

### CODE REVIEW

File: github/ci/prow-deploy/kustom/components/docker-mirror-proxy/pv/patches/JsonRFC6902/deployment.yaml [hunk 1]
```diff
@@ -11,3 +11,8 @@
     name: storage
     persistentVolumeClaim:
       claimName: docker-mirror-proxy
+- op: add
```
Comment (dhiller) [inline github/ci/prow-deploy/kustom/components/docker-mirror-proxy/pv/patches/JsonRFC6902/deployment.yaml:14]: Why is that necessary now?

Comment (brianmcarey) [reply github/ci/prow-deploy/kustom/components/docker-mirror-proxy/pv/patches/JsonRFC6902/deployment.yaml:14]: By default the new openshift cluster runs the pods as a non privileged user - the docker-mirror-proxy pod ends up in a error as it tries to access files under /etc

https://github.com/rpardini/docker-registry-proxy/blob/51585675e6543f6e9b81a2b15b15e8f0778119e2/entrypoint.sh#L7

Comment (dhiller) [reply github/ci/prow-deploy/kustom/components/docker-mirror-proxy/pv/patches/JsonRFC6902/deployment.yaml:14]: Thanks for the explanation!

**Review by dhiller — COMMENTED** [review]
/approve

**Review by dhiller — APPROVED** [review]
